src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -571,12 +574,14 @@ export default function parseManifest(
           representationMimeTypes,
           adaptationCodecs || null,
           representationCodecs,
-          adaptationChildren.role || null
+          adaptationChildren.roles || null
         );
 
-        const isMainAdaptation = !!adaptation.children.role &&
-          adaptation.children.role.value === "main" &&
-          adaptation.children.role.schemeIdUri === "urn:mpeg:dash:role:2011";
+        const { roles } = adaptation.children;
```

Comment:
``roles`` is used just before through the ``adaptationChildren`` object, which is a reference to ``adaptation.children``. So, this line looks like it comes from another function.

(well, it's true that this function is too HUGE to be readable)